geo/src/algorithm/centroid.rs

```diff
@@ -244,6 +244,40 @@ where
     }
 }
 
+///
+/// ```
+/// use geo::{MultiPoint, Point};
+/// use geo::algorithm::centroid::Centroid;
+///
+/// let empty: Vec<Point<f64>> = Vec::new();
+/// let empty_multi_points: MultiPoint<_> = empty.into();
+/// assert_eq!(empty_multi_points.centroid(), None);
+///
+/// let points: MultiPoint<_> = vec![(5., 1.), (1., 3.), (3., 2.)].into();
+/// assert_eq!(points.centroid(), Some(Point::new(3., 2.)));
+/// ```
+///
+impl<T> Centroid<T> for MultiPoint<T>
+where
+    T: Float,
+{
+    type Output = Option<Point<T>>;
+
+    fn centroid(&self) -> Self::Output {
+        if self.0.is_empty() {
+            return None;
+        }
+        let sum = self.0.iter().fold(
+            Point::new(T::zero(), T::zero()),
+            |a: Point<T>, b: &Point<T>| Point::new(a.x() + b.x(), a.y() + b.y()),
```

Comment:
do we need `b` to be **`&`**`Point<T>`?

Reply:
`.iter()` returns a `Iterator<Item=&Point<T>>` thus you can't really do any other thing (type inference might work here).